gtsfm/frontend/retriever/sequential_retriever.py

```diff
@@ -0,0 +1,43 @@
+"""Default image retriever for front-end.
+Authors: Travis Driver, Jon Womack
+"""
+from typing import List, Tuple
+
+import dask
+from dask.delayed import Delayed
+
+from gtsfm.frontend.retriever.retriever_base import RetrieverBase
+import gtsfm.utils.logger as logger_utils
+
+
+logger = logger_utils.get_logger()
+
+
+class SequentialRetriever(RetrieverBase):
+    """Class for sequential image retrieval. Keeps all image pairs from Dataloader."""
+
+    def __init__(self, image_pair_indices):
```

Comment:
Thanks for all the updates, Jonathan. we are missing a type hint here for `image_pair_indices`.

I think of image pair indices as the output, so I'm a bit confused why it's an input to the constructor.

Reply:
Our plan is to have the Retriever return pairs of image ids, determined by a list of images ids provided by the Dataloader. However, the loaders currently have a number of loader specific methods that provide/validate image pairs.

I could pass the List of image_ids and the DataLoader to the retriever. But this would require passing the Dataloader to scene_optimizer.compute_computation_graph(), and then the initialization of the Retriever. Then, the retriever could get image_id pairs using both the loader's logic (is_valid_pair()/get_valid_pairs()) and the retriever's logic (vocab, etc).

Is there an alternative approach that is preferable?

Reply:
(I'll add a type hint for whichever method we end up choosing)

Reply:
Hi Jonathan, thanks for the comments here. I think we should focus on which API design is the cleanest. Let's just pass the loader to the run function of the SceneOptimizer, so we have access to it. Retriever should be creating the image pairs, not accepting them.